Tests/iaas/volume-backup/volume-backup-tester.py

```diff
@@ -257,25 +310,40 @@ def main():
     )
     args = parser.parse_args()
     openstack.enable_logging(debug=args.debug)
+    logging.basicConfig(
+        format="%(levelname)s: %(message)s",
+        level=logging.DEBUG if args.debug else logging.INFO,
+    )
 
     # parse cloud name for lookup in clouds.yaml
     cloud = os.environ.get("OS_CLOUD", None)
     if args.os_cloud:
         cloud = args.os_cloud
-    assert cloud, (
-        "You need to have the OS_CLOUD environment variable set to your "
-        "cloud name or pass it via --os-cloud"
-    )
+    if not cloud:
+        raise Exception(
+            "You need to have the OS_CLOUD environment variable set to your "
+            "cloud name or pass it via --os-cloud"
+        )
     conn = connect(
         cloud,
         password=getpass.getpass("Enter password: ") if args.ask else None
     )
     if args.cleanup_only:
```

Comment:
I would at least consider the following simplification.

Pull out the 'then' branch and perform it unconditionally, followed by `if args.clean_only: return`. Then go ahead with the try block below.

Reasoning: simplify control flow. Also: I'm not sure we should proceed with the test if we can't even clean up properly.

Reply:
Actually, the reasoning is to reduce duplication of code... but also kind-of simplify control flow

Reply:
Done. The initial cleanup now also throws an exception and aborts in case of errors in a regular run.